### PR TITLE
Update cardano-node version to 8.0.0

### PR DESCRIPTION
**Pull request by Unisay**
- [x] I updated `cardano-node` flake input to refer to the version `8.0.0`.
- [x] Updated hackageNix, iohkNix, CHaP to latest versions.
- [x] Removed `source-repository-package` for `cardano-node` in order to get the latest one from CHaP

### Issue Numbers

ADP-2956
https://github.com/input-output-hk/cardano-wallet/issues/3945

### CODE REVIEW

**Comment by Unisay** [comment]
When attempting to connect the `cardano-wallet` compiled with the node version `1.35.4` to a newer node (`8.0.0`) this error shows up:
```
DecoderFailure
 (LocalStateQuery HardForkBlock 
    (': * ByronBlock (': * (ShelleyBlock (TPraos StandardCrypto) (ShelleyEra StandardCrypto)) 
    (': * (ShelleyBlock (TPraos StandardCrypto) (ShelleyMAEra 'Allegra StandardCrypto)) 
    (': * (ShelleyBlock (TPraos StandardCrypto) (ShelleyMAEra 'Mary StandardCrypto)) 
    (': * (ShelleyBlock (TPraos StandardCrypto) (AlonzoEra StandardCrypto)) 
    (': * (ShelleyBlock (Praos StandardCrypto) (BabbageEra StandardCrypto)) 
    (': * (ShelleyBlock (Praos StandardCrypto) (ConwayEra StandardCrypto)) ('[] *)))))))
    ) 

    Query
    (BlockQuery (HardForkBlock 
    (': * ByronBlock (': * (ShelleyBlock (TPraos StandardCrypto) (ShelleyEra StandardCrypto)) 
    (': * (ShelleyBlock (TPraos StandardCrypto) (ShelleyMAEra 'Allegra StandardCrypto)) 
    (': * (ShelleyBlock (TPraos StandardCrypto) (ShelleyMAEra 'Mary StandardCrypto)) 
    (': * (ShelleyBlock (TPraos StandardCrypto) (AlonzoEra StandardCrypto)) 
    (': * (ShelleyBlock (Praos StandardCrypto) (BabbageEra StandardCrypto)) 
    (': * (ShelleyBlock (Praos StandardCrypto) (ConwayEra StandardCrypto)) ('[] *))))))))))
 )
 ServerAgency TokQuerying BlockQuery
    (QueryIfCurrent (QS (QS (QS (QS (QS (QZ GetCurrentPParams))))))) 
    (DeserialiseFailure 53 "expected integer")


```

File: lib/wallet/cardano-wallet.cabal [hunk 1]
```diff
@@ -147,11 +147,9 @@ library
     , OddWord
     , optparse-applicative
     , ouroboros-consensus
-    , ouroboros-consensus-byron
```
Comment (Unisay) [inline lib/wallet/cardano-wallet.cabal:150]: image

File: cabal.project [hunk 2]
```diff
@@ -50,11 +50,11 @@ repository cardano-haskell-packages
     d4a35cd3121aa00d18544bb0ac01c3e1691d618f462c46129271bccf39f7e8ee
 
 -- repeating the index-state for hackage to work around hackage.nix parsing limitation
-index-state: 2023-03-06T05:24:58Z
+index-state: 2023-06-06T00:00:00Z
```
Comment (jonathanknowles) [inline cabal.project:53]: I suspect this needs to be the same as the one on line 57:
```
  , cardano-haskell-packages 2023-06-05T06:39:32Z
```

**Comment by Unisay** [comment]
Superseded by https://github.com/cardano-foundation/cardano-wallet/pull/3996

**Pull request closed without merging**